File: Year-2019/Mar/2019-03-08-125/watercow/125.c
class Solution {

public:
	bool isPalindrome(string s) {
		int length = s.length();
		for (int i = 0, j = length - 1; i < j;) {
			if (!isalpha(s[i]) && !isdigit(s[i])) i++;
			else if (!isalpha(s[j]) && !isdigit(s[j])) j--;
			else if (tolower(s[i]) != tolower(s[j])) return false;
			else {
				i++; j--;
			}
		}
		return true;
	}
};

File: Year-2019/Mar/2019-03-08-125/lb/125.java
class Solution {
//双指针查询
    public boolean isPalindrome(String s) {
        s = s.toLowerCase();
        char[] sc = s.toCharArray();
        int length = s.length()-1;
        for (int i=0;i<length;){
        //忽略除符号数字以外的所有字符
            if ((sc[i]<'a' || sc[i]>'z' ) && (sc[i]<'0' || sc[i]>'9')){
                i++;
                continue;
            }
            if((sc[length]<'a' || sc[length]>'z') && (sc[length]<'0' || sc[length]>'9')){
                length--;
                continue;
            }
            if (sc[i] != sc[length])
                return false;
            i++;
            length--;
        }
        return true;
    }
}
